nuxeo-routing-core/src/main/java/org/nuxeo/ecm/platform/routing/core/impl/GraphNodeImpl.java

```diff
@@ -856,25 +865,26 @@ public void cancelSubRoute() throws DocumentRouteException {
     @Override
     public List<EscalationRule> evaluateEscalationRules() {
         List<EscalationRule> rulesToExecute = new ArrayList<EscalationRule>();
-        OperationContext context = getExecutionContext(getSession());
         // add specific helpers for escalation
         for (EscalationRule rule : getEscalationRules()) {
-            Expression expr = new RoutingScriptingExpression(rule.condition,
-                    new RoutingScriptingFunctions(context, rule));
-            Object res = null;
-            try {
-                res = expr.eval(context);
-            } catch (RuntimeException e) {
+            try (OperationContext context = getExecutionContext(getSession())) {
+                Expression expr = new RoutingScriptingExpression(rule.condition,
+                        new RoutingScriptingFunctions(context, rule));
+                Object res = expr.eval(context);
+                if (!(res instanceof Boolean)) {
+                    throw new DocumentRouteException("Condition for rule " + rule + " of node '" + getId()
+                            + "' of graph '" + graph.getName() + "' does not evaluate to a boolean: " + rule.condition);
+                }
+                boolean bool = Boolean.TRUE.equals(res);
+                if ((!rule.isExecuted() || rule.isMultipleExecution()) && bool) {
+                    rulesToExecute.add(rule);
+                }
+            } catch (RuntimeException|OperationException e) {
+                if (e instanceof DocumentRouteException) {
```

Comment:
Formatting: space before/after `|`.
Catch/rethrow `DocumentRouteException` in a separate `catch` block.